### PR TITLE
Add `cancel-wiki-edit-button` feature

### LINKED ISSUE #4460: `Cancel` button when editing wiki pages
When editing a page on a GitHub wiki, there is no intuitive way to stop editing and discarding the changes. 
Unlike editing e.g. issue titles or issue text. (Note that the first shows a link, the second has a red-label button)

The only cancel-ish thing users see is a red-label `Delete` button which could easily be confused with the same looking `Cancel` button known from issues already:
wiki
(To remove the danger here, the delete one could have a trash can icon as well.)

### PR DESCRIPTION
**Pull request by kidonng**
Close #4460

## Test URLs

- New page: https://github.com/tooomm/wikitest/wiki/_new
- Edit page: https://github.com/tooomm/wikitest/wiki/Getting-Started/_edit

## Screenshot

Screen Shot 2021-06-16 at 20 49 58

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -384,6 +384,7 @@ Thanks for contributing! 🦋🙌
 - [](# "linkify-full-profile-readme-title") [Linkifies the readme text on profile pages.](https://user-images.githubusercontent.com/16872793/90910173-ebe4bf00-e3a4-11ea-8fc5-aea3d1a2e5e5.png)
 - [](# "convert-pr-to-draft-improvements") [Moves the "Convert PR to Draft" button to the mergeability box and adds visual feedback to its confirm button.](https://user-images.githubusercontent.com/1402241/95644892-885f3f80-0a7f-11eb-8428-8e0fb0c8dfa5.gif)
 - [](# "clean-header-search-field") [Clears duplicate queries in the header search field.](https://user-images.githubusercontent.com/1402241/114177338-7c890300-9966-11eb-83a3-a711a76fae99.png)
+- [](# "cancel-edit-button") [Adds a cancel button when editing wiki pages.](https://user-images.githubusercontent.com/44045911/121776611-6abf1880-cbc0-11eb-9463-b805a5aa81f8.png)
```
Comment (kidonng) [inline readme.md:387]: I have no idea where to put it, but it looks like a GitHub shortcoming so...

**Comment by kidonng** [comment]
@fregante Do you think it worth adding something like `isEditingWiki` to `github-url-detection`? This should also help with #4463

File: source/features/cancel-wiki-edit-button.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+async function init(): Promise<void> {
+	(await elementReady('#gollum-editor-submit'))!.after(
+		<a
+			className="flex-auto btn btn-danger"
+			href={location.href.replace(/\/_(edit|new)$/, '')}
```
Comment (kidonng) [inline source/features/cancel-wiki-edit-button.tsx:11]: There is another way to do get the previous page: `select('[name="gollum-editor"]').getAttribute('action')`, just FYI.

**Comment by yakov116** [comment]
Shouldn't the cancel button go on the other side? (Left side)

**Comment by kidonng** [comment]
I think we should avoid layout changes if possible?

**Comment by yakov116** [comment]
> I think we should avoid layout changes if possible?

Not sure what you mean by that.

RG Has a feature that moves all `close` buttons to the left to avoid a mistaken click.
image

**Comment by kidonng** [comment]
I got it. Thought you were saying I should swap the buttons.

**Review by yakov116 — APPROVED** [review]
LGTM

File: readme.md [hunk 3]
```diff
@@ -384,6 +384,7 @@ Thanks for contributing! 🦋🙌
 - [](# "linkify-full-profile-readme-title") [Linkifies the readme text on profile pages.](https://user-images.githubusercontent.com/16872793/90910173-ebe4bf00-e3a4-11ea-8fc5-aea3d1a2e5e5.png)
 - [](# "convert-pr-to-draft-improvements") [Moves the "Convert PR to Draft" button to the mergeability box and adds visual feedback to its confirm button.](https://user-images.githubusercontent.com/1402241/95644892-885f3f80-0a7f-11eb-8428-8e0fb0c8dfa5.gif)
 - [](# "clean-header-search-field") [Clears duplicate queries in the header search field.](https://user-images.githubusercontent.com/1402241/114177338-7c890300-9966-11eb-83a3-a711a76fae99.png)
+- [](# "cancel-wiki-edit-button") [Adds a cancel button when editing wiki pages.](https://user-images.githubusercontent.com/44045911/121776611-6abf1880-cbc0-11eb-9463-b805a5aa81f8.png)
```
[the remark this answers is not in the document]
Comment (kidonng) [reply readme.md]: You are commenting on the old code

File: source/features/cancel-wiki-edit-button.tsx (hunk 2 above)
Comment (kidonng) [reply source/features/cancel-wiki-edit-button.tsx]: Done

File: readme.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fregante) [reply readme.md]: Indeed I wrote this review a few days earlier but did not submit it 😅

**Comment by fregante** [comment]
Thank you @kidonng!

**Pull request merged**